superset-frontend/src/SqlLab/components/ResultSet.tsx

```diff
@@ -87,6 +122,16 @@ export default class ResultSet extends React.PureComponent<
       searchText: '',
       showExploreResultsButton: false,
       data: [],
+      showSaveDatasetModal: false,
+      newSaveDatasetName: `${props.query.tab} ${moment().format(
+        'MM/DD/YYYY HH:mm:ss',
+      )}`,
+      userDatasetsOwned: [],
+      saveDatasetRadioBtnState: DatasetRadioState.SAVE_NEW,
+      shouldOverwriteDataSet: false,
+      datasetToOverwrite: {},
+      saveModalAutocompleteValue: '',
+      userDatasetOptions: [],
```

Comment:
I feel you many be passing too many states between components unnecessarily. Can these be combined into one construct as the modal's local states and only be passed up to the parent component when users hit "Save"?